lib/teleterm/daemon/daemon.go

```diff
@@ -264,6 +284,15 @@ func (s *Service) FindGateway(gatewayURI string) (*gateway.Gateway, error) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
+	gateway, err := s.findGateway(gatewayURI)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return gateway, nil
+}
+
+func (s *Service) findGateway(gatewayURI string) (*gateway.Gateway, error) {
```

Comment:
I know this is a private method, but I think we should add a godoc stating that the caller must hold the lock
It's not "required" but I think it's worthwhile in this case

Reply:
Good point, I already added such comments to other private methods of this struct. Addressed in 720c798.